### PR TITLE
feat: first widget-tests

### PR DESCRIPTION
**Pull request by luis-stumpf**

### CODE REVIEW

**Comment by luis-stumpf** [comment]
@SamuelBehrmann pls mal abchecken wo ich // sam kommentiert hab da sind sachen die ich einfach ned zum laufen gebracht hab

File: app/test/ui/screens/search/search_view_test.dart [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    );
+
+    CustomAppBar customAppBar =
+        tester.widget<CustomAppBar>(find.byType(CustomAppBar).first);
+
+    expect(customAppBar.title, 'Suche');
+    expect(find.byType(CustomAppBar), findsOneWidget);
+    expect(find.byIcon(Icons.arrow_back), findsOneWidget);
+
+    await tester.tap(find.byIcon(Icons.arrow_back));
```
Comment (SamuelBehrmann) [inline app/test/ui/screens/search/search_view_test.dart:42]: **issue:** Separate test

File: app/test/ui/widgets/custom_card_test.dart [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  testWidgets('CustomCard renders correctly', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          body: customCard,
+        ),
+      ),
+    );
+
+    expect(find.byType(CustomCard), findsOneWidget);
```
Comment (SamuelBehrmann) [inline app/test/ui/widgets/custom_card_test.dart:21]: **suggestion:** When testing the widget you dont need to check if its there just check for actual properties.

File: app/test/ui/widgets/custom_search_bar_test.dart [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      (WidgetTester tester) async {
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          body: customSearchBarWithoutQuery,
+        ),
+      ),
+    );
+
+    await tester.enterText(find.byType(TextField), 'lunge');
```
Comment (SamuelBehrmann) [inline app/test/ui/widgets/custom_search_bar_test.dart:41]: **issue:** you should not need to enter a text for initial query

File: app/test/ui/widgets/custom_search_bar_test.dart [hunk 4]
```diff
@@ -0,0 +1,60 @@
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:medi_support/ui/widgets/custom_search_bar.dart';
+
+void main() {
+  CustomSearchBar customSearchBarWithoutQuery = CustomSearchBar(
```
Comment (SamuelBehrmann) [inline app/test/ui/widgets/custom_search_bar_test.dart:7]: **suggestion:** Move into the tests, the test should not be dependet on a variable outside the test. exceptions are variables reset by "setup(...)"

File: app/test/ui/widgets/custom_text_field_test.dart [hunk 5]
```diff
@@ -0,0 +1,65 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:medi_support/ui/widgets/custom_text_field.dart';
+
+void main() {
+  final Completer<void> completerSubmit = Completer<void>();
```
Comment (SamuelBehrmann) [inline app/test/ui/widgets/custom_text_field_test.dart:8]: **issue:** We should never need these

File: app/test/ui/widgets/custom_text_field_test.dart [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  final Completer<void> completerOnTabOutside = Completer<void>();
+  CustomTextField customTextField = CustomTextField(
+    onSubmitted: completerSubmit.complete,
+    onTapOutside: completerOnTabOutside.complete,
+  );
+  testWidgets('CustomTextField renders correctly', (WidgetTester tester) async {
```
Comment (SamuelBehrmann) [inline app/test/ui/widgets/custom_text_field_test.dart:14]: **suggestion:**  remove test

File: app/test/ui/widgets/custom_text_field_test.dart [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+        home: Scaffold(
+          body: customTextField,
+        ),
+      ),
+    );
+
+    expect(find.byType(CustomTextField), findsOneWidget);
+  });
+
+  testWidgets('CustomTextField renders correctly whit onSubmitted working',
```
Comment (SamuelBehrmann) [inline app/test/ui/widgets/custom_text_field_test.dart:26]: **suggestion:** 
```suggestion
  testWidgets('CustomTextField renders correctly with onSubmitted working',
```

and extract into separate test. Call backvalidation should be tested separate

File: app/test/ui/widgets/custom_text_field_test.dart [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    tester.firstWidget(find.byType(CustomTextField));
+
+    const String testText = 'Test input';
+    await tester.enterText(find.byType(TextField), testText);
+    await tester.testTextInput.receiveAction(TextInputAction.done);
+
+    expect(completerSubmit.isCompleted, isTrue);
+  });
+
+  testWidgets('CustomTextField renders correctly whit onTabOutside working',
```
Comment (SamuelBehrmann) [inline app/test/ui/widgets/custom_text_field_test.dart:45]: **suggestion:** 
See previous 
```suggestion
  testWidgets('CustomTextField renders correctly with onTabOutside working',
```

File: app/test/ui/widgets/custom_text_form_field_test.dart [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+              hint: '',
+            ),
+          ),
+        ),
+      );
+
+      expect(find.text(initialText), findsOneWidget);
+    });
+
+    testWidgets('Toggles password visibility', (WidgetTester tester) async {
```
Comment (SamuelBehrmann) [inline app/test/ui/widgets/custom_text_form_field_test.dart:24]: **praise:** Nice test!

File: app/test/ui/widgets/custom_text_form_field_test.dart [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+      );
+
+      expect(find.byIcon(Icons.visibility_off), findsOneWidget);
+      await tester.tap(find.byIcon(Icons.visibility_off));
+      await tester.pump();
+
+      expect(find.byIcon(Icons.visibility), findsOneWidget);
+    });
+
+    testWidgets('OnChanged is working', (WidgetTester tester) async {
```
Comment (SamuelBehrmann) [inline app/test/ui/widgets/custom_text_form_field_test.dart:43]: **praise:** Good idea!

File: app/test/ui/widgets/icon_row_test.dart [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+              },
+            ),
+          ),
+        ),
+      );
+
+      await tester.tap(find.byIcon(Icons.home));
+      expect(pressedIconLabel, 'Home');
+
+      await tester.tap(find.byIcon(Icons.settings));
```
Comment (SamuelBehrmann) [inline app/test/ui/widgets/icon_row_test.dart:57]: **suggestion:** you can remove one, if one works all work

File: app/test/ui/widgets/message_test.dart [hunk 12]
```diff
@@ -0,0 +1,65 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:medi_support/ui/widgets/message.dart'; // Adjust the import according to your file structure
```
Comment (SamuelBehrmann) [inline app/test/ui/widgets/message_test.dart:3]: **suggestion:**  remove comments. applies to whole file

File: app/test/ui/widgets/message_test.dart
Comment (SamuelBehrmann) [inline app/test/ui/widgets/message_test.dart:1]: **praise:** nice one

File: app/test/ui/widgets/post_preview_test.dart [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+      ),);
+
+      final Finder titleFinder = find.text('Test Title');
+      final Finder contentFinder = find.text('Test content goes here.');
+
+      expect(titleFinder, findsOneWidget);
+      expect(contentFinder, findsOneWidget);
+
+      final Text titleWidget = tester.widget(titleFinder) as Text;
+      final Text contentWidget = tester.widget(contentFinder) as Text;
```
Comment (SamuelBehrmann) [inline app/test/ui/widgets/post_preview_test.dart:72]: **suggestion:** Dont use type cast. use type parameter instead. E.g. below:
```suggestion
      final Text contentWidget = tester.widget<Text>(contentFinder);
```

File: app/test/ui/widgets/post_preview_test.dart [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+            },
+          ),
+        ),
+      ),);
+
+      await tester.tap(find.byType(FilledButton));
+      await tester.pump(); // Trigger a frame
+
+      expect(tappedPostId, expectedPostId,
+          reason:
```
Comment (SamuelBehrmann) [inline app/test/ui/widgets/post_preview_test.dart:101]: **suggestion:** Never done it any where else, dont start now. Remove it :)

**Review by SamuelBehrmann — APPROVED** [review]

**Pull request merged**